### DISCUSSION TITLE
equivalent feature like window.prompt

### DISCUSSION THREAD

**Discussion by aboveyunhai** [chakra-ui/chakra-ui#4134] [Ideas]
# 🚀 Feature request

I wonder does Chakra have similar component to simulate `window.prompt()`;

## 🧱 Problem Statement / Justification

I know I can use `modal` combine with `useState` to simulate the `window.prompt()` behavior;
but it can be very handy I can do something like the following;

```js

const prompt = usePrompt(// customize prompt panel here, pass value into panel);

const result = window.prompt() or promprt.trig() ;
if(result) ...
```

## ✅ Proposed solution or API

Maybe it can be an extra feature from current modal?

## ↩️ Alternatives

> N/A

## 📝 Additional Information

> N/A

**Comment by anubra266** [comment]
Have you checked [Alert Dialogs](https://chakra-ui.com/docs/overlay/alert-dialog)?

**Comment by aboveyunhai** [comment]
> Have you checked [Alert Dialogs](https://chakra-ui.com/docs/overlay/alert-dialog)?

Please read my question carefully. If you ever read the doc, `Alert Dialogs` is indeed `modal`, unless I missed something from the doc,

**Comment by anubra266** [comment]
> Please read my question carefully. If you ever read the doc, `Alert Dialogs` is indeed `modal`, unless I missed something from the doc,

I read your question, and I know it's a modal, But the dialogs are streamlined to serve this purpose, while preserving accessibility.  If not, you're basically asking for JSX in a hook.

**Comment by with-heart** [comment]
This is an interesting idea! Sort of like `useToast`, but for quickly collecting user input from a dialog in a standardized way.

We've had a request before for something similar in #1554. It was closed at the time, but personally I would love it if someone explored this idea a little more because the standard way of handling this can be too heavy on boilerplate.

**Reply by anubra266** [reply]
Hmm...Basically something that handles all the work, and just returns the value.

**Reply by with-heart** [reply]
Yeap, exactly! Creating a nice little DX around something that would otherwise be quite repetitive.

**Reply by anubra266** [reply]
Sounds good, but we can't follow the above method, because a hook shouldn't render JSX.

**Reply by anubra266** [reply]
Just thought of something...we have a `PromptDialog` Component. Then export a hook, that exposes `getPromptDialogProps`; an object that exposes the states of the `DIalog`, It could also expose a `usePrompt` hook that then returns the value of the prompt`. 
Of course there is still a lot of deep stuff, but just a shallow concept.

**Reply by with-heart** [reply]
For sure. It'd have to be like `useToast` where you define the toast first, and then trigger it. Something to play with to figure out the best way to handle it.
